build/package/windows/install.ps1

```diff
@@ -46,17 +105,59 @@ Write-Host "Installing $versionData"
 
 Write-Host "Creating New Relic Agent Control directories..."
 
-[System.IO.Directory]::CreateDirectory("$acDir") | Out-Null
+[System.IO.Directory]::CreateDirectory("$acProgramFilesDir") | Out-Null
 [System.IO.Directory]::CreateDirectory("$acDataDir") | Out-Null
+Set-RestrictedAcl -Path $acProgramFilesDir
+Set-RestrictedAcl -Path $acDataDir
 [System.IO.Directory]::CreateDirectory("$acLogsDir") | Out-Null
 [System.IO.Directory]::CreateDirectory("$acLocalConfigDir") | Out-Null
 
 Write-Host "Copying New Relic Agent Control program files..."
-Copy-Item -Path ".\newrelic-agent-control.exe" -Destination "$acDir"
+Copy-Item -Path ".\newrelic-agent-control.exe" -Destination "$acProgramFilesDir"
+
+# Generate configuration based on inputs
+$localConfigPath = Join-Path $acLocalConfigDir 'local_config.yaml'
+
+# Common args
+$cliArgs = @(
+    'generate-config',
+    '--output-path', $localConfigPath,
+    '--region', $Region,
+    '--agent-set', $AgentSet
+)
+
+# Private key path is provided whenever an existing key is to be used.
+# If not provided, a new key will be generated and stored in the keys directory.
+if ([string]::IsNullOrWhiteSpace($AuthPrivateKeyPath)) {
+    [System.IO.Directory]::CreateDirectory($acIdentityKeyDir) | Out-Null
+    $AuthPrivateKeyPath = $acIdentityKeyPath
+}
+
+if (-not $FleetEnabled) {
+    $cliArgs += '--fleet-disabled'
+} else {
+    if ($FleetId) { $cliArgs += @('--fleet-id', $FleetId) }
+    if ($OrganizationId) { $cliArgs += @('--organization-id', $OrganizationId) }
+    if ($AuthParentToken) { $cliArgs += @('--auth-parent-token', $AuthParentToken) }
+    if ($AuthParentClientId) { $cliArgs += @('--auth-parent-client-id', $AuthParentClientId) }
+    if ($AuthParentClientSecret) { $cliArgs += @('--auth-parent-client-secret', $AuthParentClientSecret) }
+    if ($AuthPrivateKeyPath) { $cliArgs += @('--auth-private-key-path', $AuthPrivateKeyPath) }
+    if ($AuthClientId) { $cliArgs += @('--auth-client-id', $AuthClientId) }
+}
+
+# Proxy args (optional)
+if ($ProxyUrl) { $cliArgs += @('--proxy-url', $ProxyUrl)}
+if ($ProxyCABundleFile) { $cliArgs += @('--proxy-ca-bundle-file', $ProxyCABundleFile) }
```

Comment:
Right now there are no agents, but should we add already in the service config license key and region as we are doing for the linux service?